public/app/plugins/datasource/prometheus/components/PromQueryField.test.tsx

```diff
@@ -80,30 +75,39 @@ describe('PromQueryField', () => {
         history={[]}
       />
     );
-    await Promise.resolve();
 
-    const cascader = queryField.find<RCCascader>(RCCascader);
-    cascader.simulate('click');
-    const cascaderNode: HTMLElement = cascader.instance().getPopupDOMNode();
-
-    for (const item of Array.from(cascaderNode.getElementsByTagName('li'))) {
-      expect(metrics.includes(item.innerHTML)).toBe(true);
-    }
+    const cascader = queryField.getByTestId('bc-button');
+    fireEvent.click(cascader);
+    await waitFor(() => {
+      const listNodes = screen.getAllByRole('listitem');
+      for (const node of listNodes) {
+        expect(metrics).toContain(node.innerHTML);
+      }
+    });
```

Comment:
```suggestion
   let cascader = queryField.getByTestId('bc-button');
    fireEvent.keyDown(cascader, { keyCode: 40 });
    let listNodes = await screen.findAllByRole('menuitem');
    for (const node of listNodes) {
      expect(metrics).toContain(node.innerHTML);
    }
```

A few things here:
- For some reason calling `click` on `Cascader` and `Select` works unreliably when triggering the dropdown menu, using arrow down key down works better in most cases. 
- Altho `li` has a `listitem` role by default, it seems that in `ButtonCascader` those are replaced by `menuitem`. 
- I find it easier to use `await` plus `find*` queries, which are basically a wrapper for `waitFor` + `get*`. Also normally it is enough to use `find*` for the first query and for the rest `get*` can be used since all the async operations are done by then.

Reply:
Ah, thanks for figuring this out. Super helpful!

Reply:
Great feedback Alex! 🌟